TLM/TLM/UI/Localization/LookupTable.cs

```diff
@@ -117,15 +133,58 @@ public class LookupTable {
                             lang != Translation.DEFAULT_LANGUAGE_CODE) {
                             continue;
                         }
-                        AllLanguages[lang][key] = cell;
+
+                        allLanguages[lang][key] = cell;
                     }
                 }
             }
+        }
 
-#if DUMP_TRANSLATIONS
-            DumpTranslationsToCsv();
-#endif
-            Log._Debug($"Loaded {AllLanguages.Count} different languages for {Name}");
+        /// <summary>
+        /// Read all lines, validate and join separated lines
+        /// </summary>
+        /// <param name="sr">stream to read from</param>
+        /// <returns>collection of valid translation rows</returns>
+        private static string[] ReadLines(StreamReader sr) {
+            string[] lines = sr.ReadToEnd().Split(new[] { "\n", "\r\n" }, StringSplitOptions.None);
+            return ValidateAndJoinLines(lines);
+        }
+
+        /// <summary>
+        /// Validates lines by joining separated incomplete strings with new line characters
+        /// </summary>
+        /// <param name="lines">Lines of translation file to validate</param>
+        /// <returns>Validated and joined collection of translation strings - one per row</returns>
+        private static string[] ValidateAndJoinLines(string[] lines) {
```

Comment:
What is this?
All you needed to do is change read cell algorithm, that if a `"` is open, then `\n` should be handled as part of the string.

Reply:
https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/blob/master/TLM/TLM/UI/Localization/LookupTable.cs#L164-L186

Reply:
Wait... `ReadCsvCell` is processing single line not whole file Stream. A few lines earlier whole file stream was split on lines.

Reply:
I'd rather have file read entirely as a single string, and then we already have everything we need to parse cells, just fix those algorithms handling open quotes. I do not like joining and backtracking logic, we do not need it here.

Reply:
I'll change it later today :wink: